### ISSUE TITLE
Frontend Performance

### ISSUE THREAD

**Issue by leeoniya** [grafana/grafana#32328]
Let's get smaller. Let's get faster. Let's get lazier.

Here's the production build composition via `webpack-bundle-analyzer` (PR: #32329):

![image](https://user-images.githubusercontent.com/43234/112535345-3171cc00-8d7a-11eb-969a-6e8d407afed0.png)

`vendors`, `app` and `unicons` are loaded eagerly for a typical dashboard:

![image](https://user-images.githubusercontent.com/43234/112573451-4e2df400-8dba-11eb-8a9e-4aae76945b15.png)

- everything in `vendors/packages` looks like a duplicate of things already in `vendors/node_modules`, e.g. react, immutable, lodash, rxjs.
- react-unicons is huge, do we need it all? (Issue: https://github.com/grafana/grafana/issues/29478)
- ~~Vis.js is large and only used [Inspect variables](https://grafana.com/docs/grafana/latest/variables/inspect-variable/). lazy load!~~ done by @hugohaggmark in #32341
- Slate.js can probably be lazy-loaded
- Moment js takes some space, and it's deprecated. perhaps we can replace it with something smaller (Issue: https://github.com/grafana/grafana/issues/28922), or even just rely on the native, now widely-available [Intl.DateTimeFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat). `latest.json` in `app` is from `moment-timezone`.
- d3 is almost exclusively used for Heatmap, we can likely drop it after migrating the heatmap panel to uPlot. the other use is by node-graph (maybe just d3-force), but if we use Vis.js for it instead, then it can be dropped.

Edge team is currently in the process of dropping the `apache-arrow` and `centrifuge.protobuf` js dependencies ;)

these appear to be lazily loaded, but are still quite large:

- Cloudwatch plugin (yellow, bottom-left) -- looks like aws-sdk packages its own copies of `external: ['react', '@grafana/data', '@grafana/ui', '@grafana/runtime'],`?
- Monaco editor & related things
- brace is another large code editor (Ace). do we really need both Monaco and brace?

This situation should be addressed before we can do further analysis since it completely changes the metrics on Code coverage reports, pageload timings, etc.

**Comment by leeoniya** [comment]
432.9KB of `vendors` has been lazified :)

:1st_place_medal: @hugohaggmark 

![image](https://user-images.githubusercontent.com/43234/112650670-9e886e80-8e19-11eb-9fd8-e0e625e8d95b.png)

**Comment by leeoniya** [comment]
https://github.com/grafana/grafana/pull/32444 eliminated the Cloudwatch bundle with a redundant grafana-ui copy :champagne: 

![image](https://user-images.githubusercontent.com/43234/113016888-8596fa80-9144-11eb-9fa8-4e7746d52bfd.png)

**Comment by sunker** [comment]
> #32444 eliminated the 3.88MB Cloudwatch bundle with a redundant grafana-ui copy 

Yes fixed in 7.5.2!

**Comment by leeoniya** [comment]
this is the current tally. the last bit of [maybe] low-hanging fruit is Slate & slate plugins lazification, but that can be done in a followup issue. enough has been done to close this out.

![image](https://user-images.githubusercontent.com/43234/114779378-a9109680-9d3b-11eb-84d1-df204fa07c92.png)
